modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -1393,6 +1406,273 @@ private String _testAddSystemObjectField(
+			).objectDefinitionId(
+				objectDefinition.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						expressionBuilderDefaultValue),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER))
+			).build());
+
+		_assertObjectFieldSettingDefaultValue(
+			expressionBuilderDefaultValue,
+			ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER, objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		// Update field's default value
+
+		objectField.setObjectFieldSettings(
+			Arrays.asList(
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+					_listTypeEntryKey),
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+					ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE)));
+
+		objectField = _updateCustomObjectField(objectField);
+
+		_assertObjectFieldSettingDefaultValue(
+			_listTypeEntryKey, ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE,
+			objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		objectField.setObjectFieldSettings(
+			Arrays.asList(
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+					expressionBuilderDefaultValue),
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+					ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER)));
+
+		objectField = _updateCustomObjectField(objectField);
+
+		_assertObjectFieldSettingDefaultValue(
+			expressionBuilderDefaultValue,
+			ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER, objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		_objectFieldLocalService.deleteObjectField(objectField);
+
+		// Add field as state
+
+		objectField = _addCustomObjectField(
+			new PicklistObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).listTypeDefinitionId(
+				_listTypeDefinition.getListTypeDefinitionId()
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						_listTypeEntryKey),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))
+			).required(
+				true
+			).state(
+				true
+			).build());
+
+		_assertObjectFieldSettingDefaultValue(
+			_listTypeEntryKey, ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE,
+			objectField);
+
+		_assertDefaultValue(_listTypeEntryKey, objectField);
+
+		_objectFieldLocalService.deleteObjectField(objectField);
+
+		// Try to add field as state without default value
```

Comment:
@devpedrotavares this part related to required values you can put in this part [here](https://github.com/liferay/liferay-portal/blob/95d934c3b3e3e0d5db4917039f1e8e2a4e0b12c6/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java#L844). Just below this [line](https://github.com/liferay/liferay-portal/blob/95d934c3b3e3e0d5db4917039f1e8e2a4e0b12c6/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java#L885). You can use the same pattern that already exist.